theano/sandbox/gpuarray/dnn.py

```diff
@@ -1472,6 +1475,27 @@ def local_log_softmax_dnn(node):
         return [new_softmax(softmax_node.inputs[0])]
 
 
+@register_opt('cudnn')
+@op_lifter([LogSoftmax])
+def local_logsoftmax_to_dnn(node, ctx_name):
+    if not dnn_available(ctx_name) or version() < 3:
+        # No log-softmax before cudnn v3
+        return
+
+    if (isinstance(node.op, LogSoftmax) and node.inputs[0].owner and
+            isinstance(node.inputs[0].owner.op, HostFromGpu)):
```

Comment:
op_lifter will decide when this op need to be moved to the GPU. So yo don't need any of those condition. node will be LogSoftmax and we will want to move it to the GPU. (so the input or the output is on the GPU)